components/datetime/src/format/time_zone.rs

```diff
@@ -54,258 +54,6 @@ where
-                    .short_specific_non_location_format(w, time_zone)
-                    .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?,
-                4 => time_zone_format
-                    .long_specific_non_location_format(w, time_zone)
-                    .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?,
-                _ => {
-                    return Err(Error::Pattern(PatternError::FieldLengthInvalid(
-                        FieldSymbol::TimeZone(zone_symbol),
-                    )))
-                }
-            },
-            fields::TimeZone::UpperZ => match field.length.idx() {
-                1..=3 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::Basic,
-                    IsoMinutes::Required,
-                    IsoSeconds::Optional,
-                )?,
-                4 => time_zone_format.localized_gmt_format(w, time_zone)?,
-                5 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::UtcExtended,
-                    IsoMinutes::Required,
-                    IsoSeconds::Optional,
-                )?,
-                _ => {
-                    return Err(Error::Pattern(PatternError::FieldLengthInvalid(
-                        FieldSymbol::TimeZone(zone_symbol),
-                    )))
-                }
-            },
-            fields::TimeZone::UpperO => match field.length.idx() {
-                1..=4 => time_zone_format.localized_gmt_format(w, time_zone)?,
-                _ => {
-                    return Err(Error::Pattern(PatternError::FieldLengthInvalid(
-                        FieldSymbol::TimeZone(zone_symbol),
-                    )))
-                }
-            },
-            fields::TimeZone::LowerV => match field.length.idx() {
-                1 => time_zone_format
-                    .short_generic_non_location_format(w, time_zone)
-                    .or_else(|_| time_zone_format.generic_location_format(w, time_zone))
-                    .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?,
-                4 => time_zone_format
-                    .long_generic_non_location_format(w, time_zone)
-                    .or_else(|_| time_zone_format.generic_location_format(w, time_zone))
-                    .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?,
-                _ => {
-                    return Err(Error::Pattern(PatternError::FieldLengthInvalid(
-                        FieldSymbol::TimeZone(zone_symbol),
-                    )))
-                }
-            },
-            fields::TimeZone::UpperV => match field.length.idx() {
-                1 => todo!("#606 (BCP-47 identifiers)"),
-                2 => todo!("#606 (BCP-47 identifiers)"),
-                3 => time_zone_format
-                    .exemplar_city(w, time_zone)
```

Comment:
Do we need to create a new format enum for it? Looks like this is a special formatting.

Reply:
Hmm, that's a good question. 

We decided not to expose this publicly, but the nice thing about our current design is that everything is internal. 

We can add a `.load_exemplar_cities()` function that is only `pub(super)` or `pub(crate)` (depending on which is needed), so that users cannot use it on their own, but it will be usable from `ZonedDateTimeFormat` when this pattern item is encountered.

Reply:
I created a new format enum `ExemplarCity` to address it